tools/koch/koch.nim

```diff
@@ -529,32 +531,13 @@ proc testTools(cmd: string) =
   nimexecFold("Run nimsuggest tests", "r nimsuggest/tester")
 
 
-proc runCI(cmd: string) =
-  doAssert cmd.len == 0, cmd # avoid silently ignoring
-  echo "runCI: ", cmd
-  echo hostInfo()
-  # boot without -d:nimHasLibFFI to make sure this still works
-  # `--lib:lib` is needed for bootstrap on openbsd, for reasons described in
-  # https://github.com/nim-lang/Nim/pull/14291 (`getAppFilename` bugsfor older nim on openbsd).
-  kochExecFold("Boot in release mode", "boot -d:release -d:nimStrictMode --lib:lib")
-
-  let batchParam = "--batch:$1" % "NIM_TESTAMENT_BATCH".getEnv("_")
-  buildTools()
-  ## run tests
-  nimexecFold("Test nimscript", "e tests/test_nimscript.nims")
-  when defined(windows):
-    nimexecFold("Compile tester", "c --usenimcache -d:posix --compileOnly testament/testament")
-
-  # main bottleneck here
-  # xxx: even though this is the main bottleneck, we could speedup the rest via batching with `--batch`.
-  # BUG: with initOptParser, `--batch:'' all` interprets `all` as the argument of --batch, pending bug #14343
-  nimexecFold("Run tester", "c -r --putenv:NIM_TESTAMENT_REMOTE_NETWORKING:1 -d:nimStrictMode testament/testament $# all" % batchParam)
-
-  testTools(cmd)
-
-  when not defined(bsd):
-    # the BSDs are overwhelmed already, so only run this test on the other machines:
-    kochExecFold("Boot Nim ORC", "boot -d:release --gc:orc --lib:lib")
+proc runCI() =
+  ## Builds the compiler and tooling in the context of continuous integration
+  ## (=CI).
+  # enable strict mode in CI, which turns some warnings and hints into
+  # errors. In addition, don't abort after the first error
+  kochExec("boot -d:danger -d:nimStrictMode --errorMax:3")
+  buildTools("-d:nimStrictMode --errorMax:3")
```

Comment:
Instead of this, I think having buildReleaseBinaries taking flags is a better idea. That'd allow the build target logic to stay in one place.